src/eos-autoupdater.c

```diff
@@ -140,6 +141,41 @@ update_stamp_file (void)
                      NULL);
     return;
   }
+
+  /* Set the file’s mtime to include the randomised delay. */
+  if (randomized_delay_days > 0)
+    {
+      g_autoptr(GFileInfo) file_info = NULL;
+      GTimeVal mtime;
+      gint32 actual_delay_days;
+
+      file_info = g_file_query_info (stamp_file, G_FILE_ATTRIBUTE_TIME_MODIFIED,
+                                     G_FILE_QUERY_INFO_NONE, NULL, &error);
+      if (error != NULL)
+        {
+          sd_journal_send ("MESSAGE_ID=%s", EOS_UPDATER_STAMP_ERROR_MSGID,
+                           "PRIORITY=%d", LOG_CRIT,
+                           "MESSAGE=Failed to get stamp file info: %s", error->message,
+                           NULL);
+          return;
+        }
+
+      g_get_current_time (&mtime);
+      actual_delay_days = g_random_int_range (0, randomized_delay_days + 1);
+      mtime.tv_sec += actual_delay_days * SEC_PER_DAY;
+      g_file_info_set_modification_time (file_info, &mtime);
```

Comment:
As commented on Slack, let's add a comment here that we're setting an mtime in the future.